pkg/ottl/ottlfuncs/func_get_semconv_span_name.go

```diff
@@ -0,0 +1,190 @@
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlspan"
+)
+
+func NewGetSemconvSpanNameFactory() ottl.Factory[ottlspan.TransformContext] {
+	return ottl.NewFactory("GetSemconvSpanName", nil, createGetSemconvSpanNameFunction)
+}
+
+func createGetSemconvSpanNameFunction(_ ottl.FunctionContext, _ ottl.Arguments) (ottl.ExprFunc[ottlspan.TransformContext], error) {
+	return getSemconvSpanName()
+}
+
+func getSemconvSpanName() (ottl.ExprFunc[ottlspan.TransformContext], error) {
+	return func(_ context.Context, tCtx ottlspan.TransformContext) (any, error) {
+		return SemconvSpanName(tCtx.GetSpan()), nil
+	}, nil
+}
+
+func SemconvSpanName(span ptrace.Span) string {
+	switch span.Kind() {
+	case ptrace.SpanKindServer:
+		spanName := httpSpanName(span, semconv.HTTPRouteKey)
+		if spanName != "" {
+			return spanName
+		}
+		spanName = rpcSpanName(span)
+		if spanName != "" {
+			return spanName
+		}
+		spanName = messagingSpanName(span)
+		if spanName != "" {
+			return spanName
+		}
+	case ptrace.SpanKindConsumer:
+		spanName := messagingSpanName(span)
+		if spanName != "" {
+			return spanName
+		}
+	case ptrace.SpanKindClient:
+		spanName := httpSpanName(span, semconv.URLTemplateKey)
+		if spanName != "" {
+			return spanName
+		}
+		spanName = rpcSpanName(span)
+		if spanName != "" {
+			return spanName
+		}
+		spanName = dbSpanName(span)
+		if spanName != "" {
+			return spanName
+		}
+		spanName = messagingSpanName(span)
+		if spanName != "" {
+			return spanName
+		}
+	case ptrace.SpanKindProducer:
+		spanName := messagingSpanName(span)
+		if spanName != "" {
+			return spanName
+		}
+	}
+	// If no semantic convention defines the span name, default to the original span name
+	return span.Name()
+}
+
+// https://opentelemetry.io/docs/specs/semconv/http/http-spans/
+func httpSpanName(span ptrace.Span, subject attribute.Key) string {
+	if method, ok := attributeValue(span, semconv.HTTPRequestMethodKey, "http.method"); ok {
+		if subjectVal, ok := span.Attributes().Get(string(subject)); ok {
+			return method.AsString() + " " + subjectVal.AsString()
+		}
+		return method.AsString()
+	}
+	return ""
+}
+
+// https://opentelemetry.io/docs/specs/semconv/rpc/rpc-spans/
+func rpcSpanName(span ptrace.Span) string {
+	if system, ok := span.Attributes().Get(string(semconv.RPCSystemKey)); ok {
+		method, okMethod := attributeValue(span, semconv.RPCMethodKey, "rpc.grpc.method")
+		service, okService := attributeValue(span, semconv.RPCServiceKey, "rpc.grpc.service")
+
+		if okMethod && okService {
+			return service.AsString() + "/" + method.AsString()
+		}
+		if okMethod {
+			return method.AsString()
+		}
+		if okService {
+			return service.AsString() + "/*"
+		}
+		return "(" + system.AsString() + ")"
+	}
+	return ""
+}
+
+// https://opentelemetry.io/docs/specs/semconv/database/database-spans/
+func dbSpanName(span ptrace.Span) string {
+	if system, ok := attributeValue(span, semconv.DBSystemNameKey, "db.system"); ok {
```

Comment:
```suggestion
	if system, ok := attributeValue(span, semconv.DBSystemNameKey, "db.system.name"); ok {
```

Reply:
We want to support both the new attribute `db.system.name` ([specs](https://opentelemetry.io/docs/specs/semconv/registry/attributes/db/#db-system-name), aka `semconv.DBSystemNameKey` ) end the deprected `db.system` ([specs](https://opentelemetry.io/docs/specs/semconv/registry/attributes/db/#db-system)).

The deprecated `db.system` attribute is still used by instrumentation libs like:
* https://pkg.go.dev/github.com/middleware-labs/opentelemetry-go-extra/otelsql
* https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Instrumentation.SqlClient/CHANGELOG.md
* https://docs.oracle.com/en/database/oracle/oracle-database/26/odpnt/featOpenTelemetry.html